File: cosmol_viewer/examples/render_molecule.rs
use cosmol_viewer::{Scene, Viewer, shapes::Molecule, utils::Stylable};
use std::path::Path;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mol = Molecule::from_sdf(include_str!("../examples/6fi1_ligand.sdf"))?.centered();

    let mut scene = Scene::new();

    scene.add_shape_with_id("mol", mol);

    let viewer = Viewer::render(&scene, 800.0, 500.0)?;

    let img = viewer.take_screenshot();

    img.save(Path::new("screenshot.png"))?;

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());

    Ok(())
}

File: cosmol_viewer/examples/render_molecule.py
from cosmol_viewer import Molecule, Scene, Viewer

mol_data = open("./examples/6fi1_ligand.sdf", "r", encoding="utf-8").read()

mol = Molecule.from_sdf(mol_data).centered().enable_outline(width=0.04)

scene = Scene()

scene.add_shape_with_id("molecule", mol)
scene.set_camera_view(azimuth=35, elevation=20, distance=32, fov=18)

scene.save_image("rendered_molecule.png", width=1600, height=1000)
viewer = Viewer.render(scene, width=800, height=500)

print("Press Any Key to exit...", end="", flush=True)
try:
    _ = input()
except EOFError:
    pass
